db/migrate/20231116181122_create_logs.rb

```diff
@@ -0,0 +1,26 @@
+class CreateLogs < ActiveRecord::Migration[7.0]
+  def change
+    create_table :logs do |t|
+      t.integer :owner_id
+      t.integer :coffee_id
+      t.integer :grinder_id
+      t.integer :brewer_id
+      t.text :notes
+      t.string :filter_paper
+      t.integer :dosage
+      t.integer :water_temperature
+      t.string :water_type
+      t.references :photo, null: false, foreign_key: true
```

Comment:
good job in using `references` ; you should consider doing this for all foreign keys (owner, coffee, grinder, brewer)